pkg/storage/s3/checker.go

```diff
@@ -21,7 +20,7 @@ func (s *Storage) Exists(ctx context.Context, module, version string) (bool, err
 
 	lsParams := &s3.ListObjectsInput{
 		Bucket: aws.String(s.bucket),
-		Prefix: aws.String(fmt.Sprintf("%s/@v", module)),
+		Prefix: aws.String(config.PackageVersionedName(module, version, "")),
```

Comment:
Can we use HeadObject as @linzhp suggested if it works?

Reply:
I think `HeadObject` would work here 👍 

While I am happy to incorporate that into this PR, my concern is that we would be making 2 additional requests (3 in total for `.mod`, `.zip` and `.info`  files) using `HeadObject` if my understanding is correct. 

AFAIK there's no way we can batch the 3 calls using `HeadObject`, unfortunately.

WDYT?

Reply:
Head does not scan the entire bucket like list. It does an exact match. List is charged by AWS while Head I do not think is.
If you can, have three go routines start with a wait group; I think your latency should be under a second unless AWS is doing something really wrong.

Reply:
After some research, I think you're right about the API pricing. My initial concern was more on the AWS S3 request rate limits (afaik it's at 5,500 req/s see [this doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html))

I've updated the `Exist` method for review.

Since we are only sending 3 requests in the loop, and they are not particularly computationally intensive, it is unlikely that using a Go routine would result in a significant performance improvement